Author: Luca Boccassi <luca.boccassi@gmail.com>
Description: test: skip tcti-spidev on 32bit architectures with 64bit time_t
Forwarded: https://github.com/tpm2-software/tpm2-tss/pull/2904
--- a/test/unit/tcti-spidev.c
+++ b/test/unit/tcti-spidev.c
@@ -31,6 +31,11 @@
 #include "tss2-tcti/tcti-spi-helper.h"
 #include "util/key-value-parse.h"
 
+#define LOGMODULE tests
+#include "util/log.h"
+
+#define EXIT_SKIP 77
+
 typedef enum {
     TPM_DID_VID_HEAD = 0,
     TPM_DID_VID_DATA,
@@ -208,6 +213,12 @@
 main (int   argc,
       char *argv[])
 {
+#if __SIZEOF_POINTER__ == 4 && _TIME_BITS == 64
+    // Would produce cmocka error
+    LOG_WARNING("_TIME_BITS == 64 would produce cmocka errors on this platform.");
+    return EXIT_SKIP;
+#endif
+
     const struct CMUnitTest tests[] = {
         cmocka_unit_test (tcti_spi_init_test),
     };
